### PR TITLE
Live: show popup if local changes exist

### LINKED ISSUE #36073: Unsaved changes when someone else saves a dashboard with grafana live on
**What happened**: If you modify a dashboard at the same time with someone else, your current changes get overwritten without any warning.

**What you expected to happen**: A confirmation window should pop asking me if I want the dashboard to reload.

**How to reproduce it (as minimally and precisely as possible)**: Have a dashboard open in two tabs, modify something in one but don't save, modify and save in the other.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.0.0
- Data source type & version: irrelevant
- OS Grafana is installed on: Linunx
- User OS & Browser: Firefox Linux
- Grafana plugins: irrelevant
- Others:

### PR DESCRIPTION
**Pull request by ryantxu**
Fixes: #36073

This uses configRev as a simple change tracker.  I suspect this will over estimate having local changes though :(

### CODE REVIEW

File: public/app/features/dashboard/dashgrid/DashboardGrid.tsx [hunk 1]
```diff
@@ -108,7 +108,9 @@ export class DashboardGrid extends PureComponent<Props, State> {
   };
 
   onResize: ItemCallback = (layout, oldItem, newItem) => {
-    this.panelMap[newItem.i!].updateGridPos(newItem);
+    const panel = this.panelMap[newItem.i!];
+    panel.updateGridPos(newItem);
+    panel.configRev++; // trigger change handler
```
Comment (torkelo) [inline public/app/features/dashboard/dashgrid/DashboardGrid.tsx:113]: would ne nicer if incrementing configRev was done by updateGridPos :)

Comment (ryantxu) [reply public/app/features/dashboard/dashgrid/DashboardGrid.tsx:113]: I started with that.... but everything gets at least one call to `updateGridPos`, so this was may attempt to only get the resize events :man_shrugging:

**Comment by ryantxu** [comment]
This is a baby step towards better change tracking

File: public/app/features/dashboard/dashgrid/DashboardGrid.tsx (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/dashgrid/DashboardGrid.tsx:113]: I think exposing a property/field like `configRev` is a code smell and will lead to all sorts of issues moving forward. Can we encapsulate this some way? I'm aware this PR didn't introduce it 😉

Comment (hugohaggmark) [reply public/app/features/dashboard/dashgrid/DashboardGrid.tsx:113]: Not sure it's related to this PR but `updateQueries` function in https://github.com/grafana/grafana/blob/646d95b8fbd962e8cdefd461fe69f92a05f3ce37/public/app/features/dashboard/state/PanelModel.ts#L438 will always change `configRev` for plugins that upgrade "their" queries on load. We need to solve the upgrade of queries somehow without calling updateQueries to reduce the amount of false save/change popups.

Comment (dprokop) [reply public/app/features/dashboard/dashgrid/DashboardGrid.tsx:113]: > I think exposing a property/field like configRev is a code smell and will lead to all sorts of issues moving forward. Can we encapsulate this some way? I'm aware this PR didn't introduce it 😉

+1

Comment (dprokop) [reply public/app/features/dashboard/dashgrid/DashboardGrid.tsx:113]: > I think exposing a property/field like configRev is a code smell and will lead to all sorts of issues moving forward. Can we encapsulate this some way? I'm aware this PR didn't introduce it 😉

+1

**Comment by ryantxu** [comment]
@torkelo @hugohaggmark @dprokop -- any thoughts on if we should close this or merge it?  I don't anticipate changing the approach without a larger dashboard "change tracker" revist.

**Comment by torkelo** [comment]
I will take over this, think this is a good foundation for a dashboard change tracker based on a configRev counter

**Review by torkelo — APPROVED** [review]

**Comment by torkelo** [comment]
think this looks good as is, and can go into v8.2, will continue and work on dashboard level config rev as well

**Pull request merged**